src/workerd/api/http.c++

```diff
@@ -1092,12 +1092,24 @@ jsg::Ref<Response> Response::redirect(
 
   // TODO(conform): The URL is supposed to be parsed relative to the "current setting's object's API
   //   base URL".
-  auto urlOptions = kj::Url::Options { .percentDecode = false, .allowEmpty = true };
-  auto maybeParsedUrl = kj::Url::tryParse(url, kj::Url::REMOTE_HREF, urlOptions);
-  if (maybeParsedUrl == nullptr) {
-    JSG_FAIL_REQUIRE(TypeError, kj::str("Unable to parse URL: ", url));
+  kj::StringPtr parsedUrl = nullptr;
+  if (flags.getSpecCompliantUrlRedirect()) {
+    auto maybeParsedUrl = url::URL::parse(jsg::usv(url));
+    if (maybeParsedUrl == nullptr) {
+      JSG_FAIL_REQUIRE(TypeError, kj::str("Unable to parse URL: ", url));
+    }
+    auto parsedSpecURL = kj::str(KJ_ASSERT_NONNULL(kj::mv(maybeParsedUrl)).getHref());
+    parsedUrl = parsedSpecURL;
+  } else {
+    auto urlOptions = kj::Url::Options { .percentDecode = false, .allowEmpty = true };
+    auto maybeParsedUrl = kj::Url::tryParse(url, kj::Url::REMOTE_HREF, urlOptions);
+    if (maybeParsedUrl == nullptr) {
+      JSG_FAIL_REQUIRE(TypeError, kj::str("Unable to parse URL: ", url));
+    }
+    auto parsedKJUrl = KJ_ASSERT_NONNULL(kj::mv(maybeParsedUrl)).toString();
```

Comment:
Tiny style nit: in KJ-style this would be `parsedKjUrl` (lowercase j)

Reply:
Done.